### PR TITLE
Support Importing/Exporting PEM data

### PR DESCRIPTION
**Pull request by simo5**
With this functions PEM files can be easily imported, and JWK Keys can be easily exported into PEM files.
@tiran @wrouesnel please take a look at this one, it implements #25 / #43

### CODE REVIEW

File: jwcrypto/jwk.py [hunk 1]
```diff
@@ -549,12 +550,84 @@ def import_from_pyca(self, key):
         else:
             raise InvalidJWKValue('Unknown key object %r' % key)
 
+    def import_from_pem(self, data, password=None):
+        """Imports a key from data loaded from a PEM file.
```
Comment (tiran) [inline jwcrypto/jwk.py]: I don't like the fact that the method attempts to load public and private keys. As an application developer I'd like to have more control over the key type and have dedicated methods to load either a private or a public key.

File: jwcrypto/jwk.py [hunk 2]
```diff
@@ -549,12 +550,84 @@ def import_from_pyca(self, key):
...
+
+        :param private_key: Whether the private key should be exported.
+         Defaults to `False` which means the public key is exported by default.
+        :param password(bytes): A password for wrapping the private key.
+         Defaults to False which will cause the operation to fail. To avoid
+         encryption the user must explicitly pass None, otherwise the user
+         needs to provide a password in a bytes buffer.
+        """
+        e = serialization.Encoding.PEM
+        if private_key is True:
```
Comment (tiran) [inline jwcrypto/jwk.py]: Please drop 'is True'

if private_key:

File: jwcrypto/jwk.py [hunk 3]
```diff
@@ -549,12 +550,84 @@ def import_from_pyca(self, key):
...
+            return self._get_public_key().public_bytes(encoding=e, format=f)
+
     @classmethod
     def from_pyca(cls, key):
         obj = cls()
         obj.import_from_pyca(key)
         return obj
 
+    @classmethod
+    def from_pem(cls, data, password=None):
```
Comment (tiran) [inline jwcrypto/jwk.py]: Is this the official API to create a JWK from PEM data? In that case ```import_from_pem()``` should be a private method.

Comment (simo5) [inline jwcrypto/jwk.py]: I don't think a classmethod can acces private methods (last time I tried for from_pyca I got nasty warnings).
But in either case I am fine with both being public methods.

File: jwcrypto/jwk.py (hunk 2 above)
Comment (simo5) [inline jwcrypto/jwk.py]: Won't this always match anything that is not "False" or "None", I want to make sure there is a True there and no a bytes blob with a private key for example.

File: jwcrypto/jwk.py (hunk 1 above)
Comment (simo5) [inline jwcrypto/jwk.py]: So would you like a qualifier in the function signature?
Something like: import_from_pem(self, data, type=None|"private"|"public"|"x509", password=None) ?
Where None would be the default and autodetect what's in the file, but with the ability to force to check for a specific type ?

File: jwcrypto/jwk.py [hunk 4]
```diff
@@ -405,6 +409,52 @@ def export_public(self):
...
+        It fails for a JWK that has only a public key or is symmetric.
+        """
+        if self.has_private:
+            return self._export_all()
+        raise InvalidJWKType("No private key available")
+
+    def export_symmetric(self):
+        if self.is_symmetric:
+            return self._export_all()
+        raise InvalidJWKType("Not a symmetric key")
```
Comment (tiran) [inline jwcrypto/jwk.py]: I'd prefer a different order:
```
if not self.is_symmetric:
     raise InvalidJWKType("Not a symmetric key")
return self._export_all()
```

Comment (simo5) [inline jwcrypto/jwk.py]: Why do you feel it is better this way ?

**Pull request closed without merging**